semgrep/semgrep/evaluation.py

```diff
@@ -206,6 +206,20 @@ def _evaluate_single_expression(
                 for pattern_range in ranges_for_pattern
             )
         }
+    elif expression.operator == OPERATORS.REGEX:
+        # remove all ranges that don't equal the ranges for this pattern
+        output_ranges = ranges_left.intersection(ranges_for_pattern)
+    elif expression.operator == OPERATORS.NOT_REGEX:
+        # remove the result if pattern-not-regex is within another pattern
+        output_ranges = {
+            _range
+            for _range in ranges_left
+            if not any(
+                _range.is_range_enclosing_or_eq(pattern_range)
+                or pattern_range.is_range_enclosing_or_eq(_range)
+                for pattern_range in ranges_for_pattern
+            )
+        }
     elif expression.operator == OPERATORS.WHERE_PYTHON:
```

Comment:
I refactored `WHERE_PYTHON` down since it's more similar to `METAVARIABLE_REGEX` and `METAVARIABLE_COMPARISON` :broom: